shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -134,26 +179,34 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
    * @return The data converted into a format more suitable for caching.
    */
   override def convertColumnarBatchToCachedBatch(input: RDD[ColumnarBatch],
-                                       schema: Seq[Attribute],
-                                       storageLevel: StorageLevel,
-                                       conf: SQLConf): RDD[CachedBatch] = {
-    def putOnGpuIfNeeded(batch: ColumnarBatch): ColumnarBatch = {
-      if (batch.numCols() > 0 && !batch.column(0).isInstanceOf[GpuColumnVector]) {
-        val s = StructType(
-          schema.map(a => StructField(a.name, a.dataType, a.nullable, a.metadata)))
-        val gpuCB = new GpuColumnarBatchBuilder(s, batch.numRows(), batch).build(batch.numRows())
-        batch.close()
-        gpuCB
-      } else {
-        batch
+     schema: Seq[Attribute],
+     storageLevel: StorageLevel,
+     conf: SQLConf): RDD[CachedBatch] = {
+    val rapidsConf = new RapidsConf(conf)
+    if (rapidsConf.isSqlEnabled) {
```

Comment:
You are also missing a check to see if the types are supported by the GPU.  We can have the plugin enabled, but if the types are not supported then we need to fall back.